LongestStreak.rb

```diff
@@ -0,0 +1,76 @@
+=begin 
+Create a function that takes an array of date hashes and return the "longest streak" (i.e. longest number of consecutive days in a row).
+
+Example
+longest_streak([
+  {
+    date: "2019-09-18"
+  },
+  {
+    date: "2019-09-19"
+  },
+  {
+    date: "2019-09-20"
+  },
+  {
+    date: "2019-09-26"
+  },
+  {
+    date: "2019-09-27"
+  },
+  {
+    date: "2019-09-30"
+  }
+]) ➞ 3
+Notes
+An empty array should return 0.
+The hashes are with symbol keys (i.e. get a date with [:date]).
+=end
+
+def calculate_longest_streak(array)
+  startposition = 0 
+  endposition = 0
+  array.each_with_index do |date, index|
+    
+  end
+
+end
+
+def longest_streak(array)
+  dates = []
+  if !array.empty?
+    array.each do |element|
+      element.each do |key, values|
+        dates.push(values[-2..10])
+      end
+    end
+  else
+    return 0
+  end
+  calculate_longest_streak(dates)
+end
+
+
+print longest_streak([
```

Comment:
algorithm suggestion:

1. loop through each value in the array. You will get a hash. You can read the date value like `hash['date']`. 
2. Then from there you can split the value to convert the date value to an array like this:
```
"2019-09-18".split('-') => ['2019', '09', '18']
```

Then you just need to compare that one with the next date. something similar to:
* check if the year is the same
* check if the month is the same. 
* if the month is not the same, then check if the date is the last day of the month and the next date is the first date of the next month.
* then check if same month then the next day is the next value. 

Like that you can see the longest streak then.

Reply:
@Anishluke92 if you cannot code this, could you post your step by step algorithm here. Like numbered instructions on how you would do this on pen and paper.